src/server/controllers/userController.ts

```diff
@@ -59,13 +61,13 @@ export async function resetPassword(req: Request, res: Response, next: NextFunct
       throw new ValidationError('User with email does not exist', 'validation.reset.password.user.not.exist');
     }
 
-    const newPassword = randomstring.generate();
-
-    user.password = md5(newPassword);
+    const resetPasswordId = uuid.v1();
+    user.resetPasswordId = resetPasswordId;
     await updateUser(user);
 
     // Don't wait to send the email
-    sendUserResetPasswordEmail(user, newPassword);
+    const changePasswordUrl = req.get('Referrer').replace('resetpassword', `changepassword/${user.id}/${user.resetPasswordId}`);
+    sendUserResetPasswordEmail(user, changePasswordUrl, language);
```

Comment:
Actually even better, maybe the `changePasswordUrl` should be built inside `sendUserResetPasswordEmail`  and the `baseUrl` read from `config`.